File: Algorithms/487. Max Consecutive Ones II/swift/solution.swift
class Solution {
    func findMaxConsecutiveOnes(_ nums: [Int]) -> Int {
        var prev: Int = -1
        var curr: Int = 0
        var maxLen: Int = 0

        for num in nums {
            if num == 0 {
                prev = curr
                curr = 0
            } else {
                curr += 1
            }
            maxLen = max(maxLen, curr + prev + 1)
        }
        return maxLen
    }
}

File: Algorithms/487. Max Consecutive Ones II/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function findMaxConsecutiveOnes($nums)
    {
        $prev = -1;
        $curr = 0;
        $maxLen = 0;
        foreach ($nums as $num) {
            if ($num == 0) {
                $prev = $curr;
                $curr = 0;
            } else {
                $curr++;
            }

            $maxLen = max($maxLen, $prev + 1 + $curr);
        }

        return $maxLen;
    }
}
